### PR TITLE
feat(graphql): generate types from graphql files

### PR DESCRIPTION
**Pull request by Brooooooklyn**
Initialize the `@affine/graphql` package, you can run `yarn workspace @affie/graphql gql-gen` to generate `src/graphql/index.ts` and `src/schema.ts` files.

The `src/graphql/index.ts` is generated by `packages/graphql-client/export-gql-plugin.cjs`;

A SWR request wrapper has been written and placed in `app/web/shared/gql.ts`

Queries:

```ts
import { someQuery, someQueryWithNoVars } from '@affine/graphql'

const swrResponse1 = useQuery({
  query: workspaceByIdQuery,
  variables: { id: '1' }
})

const swrResponse2 = useQuery({
  query: someQueryWithNoVars
})

```

Mutations:

```ts
import { someMutation } from '@affine/graphql'

const { trigger } = useMutation({
  mutation: someMutation
})

trigger({ name: 'John Doe' })

```

### CODE REVIEW

File: packages/graphql-client/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "@affine/graphql-client",
+  "version": "0.0.0",
+  "description": "Autogenerated GraphQL client for affine.pro",
+  "license": "MIT",
```
Comment (himself65) [inline packages/graphql-client/package.json]: not mit

File: packages/graphql-client/package.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "@affine/graphql-client",
```
Comment (forehalo) [inline packages/graphql-client/package.json:2]: why not just `@affine/graphql`, where we put all gql/schema/codegen stuffs in, and `GraphQLClient` that used to send gql request as well

File: apps/web/src/shared/gql.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    SWRConfiguration<
+      QueryResponse<Query>,
+      GraphQLError | GraphQLError[],
+      typeof fetcher<Query>
+    >,
+    'fetcher'
+  >
+): SWRResponse<QueryResponse<Query>, GraphQLError | GraphQLError[]>;
+export function useQuery(options: QueryOptions<any>, config?: any) {
+  return useSWR(options.query.id, () => fetcher(options), config);
```
Comment (promer94) [inline apps/web/src/shared/gql.ts:54]: You might also want to include `options.variables` as the `key` here.

```tsx
const swrResponse1 = useQuery({
 query: workspaceByIdQuery,
 variables: { id: '1' }
})
const swrResponse2 = useQuery({
 query: workspaceByIdQuery,
 variables: { id: '2' }
})
```
Then `swrResponse1` and `swrResponse2` could be cached individually

File: apps/web/src/shared/__tests__/gql.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+import { render } from '@testing-library/react';
+import type { Mock } from 'vitest';
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
+
+import { useMutation, useQuery } from '../gql';
+
+let fetch: Mock;
+describe('GraphQL wrapper for SWR', () => {
+  beforeEach(() => {
+    fetch = vi.fn(() =>
```
Comment (himself65) [inline apps/web/src/shared/__tests__/gql.spec.tsx:13]: if we wrap the `fetch` into `fetchAtom`. This could be easy to control the test behavior

Comment (himself65) [reply apps/web/src/shared/__tests__/gql.spec.tsx:13]: but for now it's ok

File: tsconfig.json [hunk 5]
```diff
@@ -28,6 +28,7 @@
       "@affine/env/*": ["./packages/env/src/*"],
       "@affine/utils": ["./packages/utils"],
       "@affine/workspace/*": ["./packages/workspace/src/*"],
+      "@affine/graphql": ["./packages/graphql/src/index.ts"],
```
Comment (himself65) [inline tsconfig.json]:
```suggestion
      "@affine/graphql": ["./packages/graphql/src"],
```

**Pull request merged**